WcaOnRails/app/views/delegate_reports/edit.html.erb

```diff
@@ -0,0 +1,20 @@
+<% provide(:title, "Report for #{@competition.name}") %>
+
+<%= render layout: 'nav' do %>
+  <h2>Editing Delegate Report for <%= @competition.name%></h2>
+  <%= simple_form_for(@delegate_report, url: competition_report_path(@competition.id)) do |f| %>
```

Comment:
Yes, because the url helper is `competition_report_path`, not `delegate_report_path`:

```
                          competition_report_edit GET    /competitions/:competition_id/report/edit(.:format)                         delegate_reports#edit
                               competition_report GET    /competitions/:competition_id/report(.:format)                              delegate_reports#show
                                                  PATCH  /competitions/:competition_id/report(.:format)                              delegate_reports#update
```

The only way I can think of to change these names would be to move the routes outside of the competitions resource, as follows:

```ruby
  resources :competitions, only: [:index, :show, :edit, :update, :new, :create] do
    get 'results/podiums' => 'competitions#show_podiums'
    get 'results/all' => 'competitions#show_all_results'
    get 'results/by_person' => 'competitions#show_results_by_person'

    patch 'registrations/selected' => 'registrations#do_actions_for_selected', as: :registrations_do_actions_for_selected
    post 'registrations/export' => 'registrations#export', as: :registrations_export
    get 'registrations/psych-sheet' => 'registrations#psych_sheet', as: :psych_sheet
    get 'registrations/psych-sheet/:event_id' => 'registrations#psych_sheet_event', as: :psych_sheet_event
    resources :registrations, only: [:index, :update, :create, :edit, :destroy], shallow: true
    get 'edit/registrations' => 'registrations#edit_registrations'
    get 'register' => 'registrations#register'
    get 'register-require-sign-in' => 'registrations#register_require_sign_in'
  end

  get 'competitions/:competition_id/report/edit' => 'delegate_reports#edit', as: :delegate_report_edit
  get 'competitions/:competition_id/report' => 'delegate_reports#show', as: :delegate_report
  patch 'competitions/:competition_id/report' => 'delegate_reports#update'
```

which gives the following routes:

```
                             delegate_report_edit GET    /competitions/:competition_id/report/edit(.:format)                         delegate_reports#edit
                                  delegate_report GET    /competitions/:competition_id/report(.:format)                              delegate_reports#show
                                                  PATCH  /competitions/:competition_id/report(.:format)                              delegate_reports#update
```

After fixing some other calls to `competition_report`, this works now.